### PR TITLE
Update mimimum supported Java version from 8 to 11

### LINKED ISSUE #1025: Move to Java 8 as minimal Java version
### Description

I want to move to Java 8 to leverage lambdas and Optionals for the internal configuration handling. Up to now the minimal version is Java 7.

Are there any objections to move on to Java 8 (for version 0.30 then) ?

### PR DESCRIPTION
**Pull request by chonton**
Closes #1025

### CODE REVIEW

**Comment by chonton** [comment]
Builds failed due to jacoco not working with java11.  Update of jacoco causes failure of tests, due to interference with jmockit.  JMockit does not have an upgrade path.
I'm considering a PR to move all tests to Mockito.

**Comment by rohanKanojia** [comment]
@chonton : I also prefer mockito over JMockit. In [Eclipse JKube](https://github.com/eclipse/jkube), we've plans to [migrate from Mockito](https://github.com/eclipse/jkube/issues/857) as well. 

Are you working on Mockito refactor? Shall I create an issue about it and assign it to you?

**Comment by chonton** [comment]
@rohanKanojia, please review

File: it/pom.xml [hunk 1]
```diff
@@ -15,12 +15,10 @@
 
   <groupId>io.fabric8.dmp.itests</groupId>
   <artifactId>dmp-it-parent</artifactId>
-  <version>0.40.0-SNAPSHOT</version>
   <packaging>pom</packaging>
 
   <modules>
     <module>builder</module>
-    <module>docker-compose</module>
```
Comment (rohanKanojia) [inline it/pom.xml:23]: Why is this integration test deleted?

File: samples/healthcheck/pom.xml [hunk 2]
```diff
@@ -100,24 +106,25 @@
               </build>
             </image>
             <image>
-              <alias>healthybox5</alias>
-              <name>busybox5</name>
+              <alias>healthy5</alias>
+              <name>nginx5</name>
               <build>
-                <from>busybox</from>
+                <from>nginx</from>
```
Comment (rohanKanojia) [inline samples/healthcheck/pom.xml:112]: Why have you changed base image to nginx? I think nginx image is bigger in size as compared to busybox.

File: samples/pom.xml [hunk 3]
```diff
@@ -18,10 +18,8 @@
 
   <modules>
     <module>build-cache</module>
-    <module>cargo-jolokia</module>
     <module>copy-from-container</module>
     <module>custom-net</module>
-    <module>data-jolokia</module>
```
Comment (rohanKanojia) [inline samples/pom.xml:24]: Why have we deleted these samples? Is it not possible to modify these samples to align with JDK 11?

Comment (chonton) [reply samples/pom.xml:24]: I tried with little success.  I'm not sure it's worth the effort.  What samples are developers interested in and will show off d-m-p's capabilities?  For example; Spring, Blade, Dropwizard, Micronaut, Quarkus, Jhipster, MicroProfile, Hilla

File: samples/healthcheck/pom.xml (hunk 2 above)
Comment (chonton) [reply samples/healthcheck/pom.xml:112]: I was attempting to get sample to work.  Apparently, #1101 is the problem here.  I've reverted to busybox.

File: samples/pom.xml (hunk 3 above)
Comment (chonton) [reply samples/pom.xml:24]: @rohanKanojia, are you and @rhuss ok to remove these samples?

Comment (rohanKanojia) [reply samples/pom.xml:24]: Umm, okay. Let's remove these samples.

**Comment by rohanKanojia** [comment]
@chonton : Could you please rebase this PR?

**Comment by chonton** [comment]
@rohanKanojia rebased as requested

**Comment by chonton** [comment]
@rohanKanojia ready to merge?

**Comment by rohanKanojia** [comment]
Yes, I was thinking cutting a patch release with your buildx related fixes and cut a new minor release with JDK upgrade

**Comment by rhuss** [comment]
@chonton first of all, thanks for your support ! Before we kill Java 8, I would love to discuss this a bit further though, as I'm not 100% convinced we should do that right now. See https://github.com/fabric8io/docker-maven-plugin/issues/1025#issuecomment-1199099333 for my reasoning.

**Comment by rhuss** [comment]
Sorry, the conflicts come from fixing the version number to 0.41-SNAPSHOT (was 0.41.0-SNAPSHOT which was wrong)

**Comment by rhuss** [comment]
btw, I'm totally +1 for moving all mocking code to Mockito.

**Pull request closed without merging**